### PR TITLE
feat(action): add name html property on Action

### PR DESCRIPTION
**Pull request by acateland**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
Action don't generate a name property on a button, making it harder for QA to target.

**What is the new behavior?**
If a name props is given to action, then it is used to render the name property.

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration
path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

File: packages/components/src/Actions/Action/Action.component.js [hunk 1]
```diff
@@ -99,6 +99,7 @@ function Action(props) {
 			bsStyle={style}
 			disabled={inProgress || disabled}
 			role={link ? 'link' : null}
+			name={props.name || props.label}
```
Comment (frassinier) [inline packages/components/src/Actions/Action/Action.component.js]: WAT

> Name of the element. For example used by the server to identify the fields in form submits.

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

File: packages/components/src/Actions/Action/Action.component.js (hunk 1 above)
Comment (acateland) [inline packages/components/src/Actions/Action/Action.component.js]: label could be used to do exactly that by default

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

File: output/components.eslint.txt [hunk 2]
```diff
@@ -8,6 +8,9 @@ Scoping to packages that match 'react-talend-components'
 /home/travis/build/Talend/ui/packages/components/src/ActionBar/ActionBar.component.js
   22:4  error  Expected space(s) after "switch"  keyword-spacing
 
+/home/travis/build/Talend/ui/packages/components/src/Actions/Action/Action.test.js
+  112:4  error  Missing semicolon  semi
```
Comment (jmfrancois) [inline output/components.eslint.txt]: please fix it

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

File: packages/forms/README.md [hunk 3]
```diff
@@ -68,6 +68,7 @@ Each action accept the following properties :
 | label                 | string                  | yes      | -
 | link                  | bool                    | no       | -
 | model                 | object                  | no       | -
+| name                  | string                  | no       | - | render a name html property
```
Comment (jmfrancois) [inline packages/forms/README.md]: render a name button html property

File: packages/components/src/Actions/Action/Action.component.js [hunk 4]
```diff
@@ -121,6 +121,7 @@ Action.propTypes = {
 	label: PropTypes.string.isRequired,
 	link: PropTypes.bool,
 	model: PropTypes.object, // eslint-disable-line react/forbid-prop-types
+	name: PropTypes.string,
```
Comment (jmfrancois) [inline packages/components/src/Actions/Action/Action.component.js]: why this is needed ?
For me a proptypes is needed only if used in the render.
if the props is passed by spread it should not be here.

Comment (acateland) [inline packages/components/src/Actions/Action/Action.component.js]: i think leaving the proptypes here could be usefull as api documentation.

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://350.talend.surge.sh)

**Pull request merged**